src/Model/CodeModelGo.cs

```diff
@@ -137,10 +137,19 @@ public IEnumerable<string> ModelImports
                 {
                     imports.Add(PrimaryTypeGo.GetImportLine("github.com/Azure/go-autorest/autorest"));
                 }
-                if (ModelTypes.Any(mt => mt is FutureTypeGo))
+                var futureModels = ModelTypes.Where(mt => mt is FutureTypeGo).Cast<FutureTypeGo>();
+                if (futureModels.Any())

```

Comment:
not sure what was wrong with the previous if, also why do we cast to FutureTypeGo

Reply:
oh ok nevermind we need the collection later, but do we need the cast ?

---
In reply to: [192249868](https://github.com/Azure/autorest.go/pull/130#discussion_r192249868) [](ancestors = 192249868)

Reply:
Yes, the ```IsDefaultReturnType``` property is only on ```FutureTypeGo```.